pkg/driver/controller_modify_volume.go

```diff
@@ -224,6 +236,33 @@ func parseModifyVolumeParameters(params map[string]string) (*modifyVolumeRequest
 	if err := validateExtraTags(addTags, false); err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, "Invalid tag value: %v", err)
 	}
+	for k, v := range noValidationTags {
+		addTags[k] = v
+	}
 	options.modifyTagsOptions.TagsToAdd = addTags
 	return &options, nil
 }
+
+func matchIOPStoIopsPerGbRatio(c cloud.Cloud, ctx context.Context, volumeID string, req *modifyVolumeRequest) (int32, error) {
+	disk, err := c.GetDiskByID(ctx, volumeID)
+	if err != nil {
+		if errors.Is(err, cloud.ErrNotFound) {
+			return 0, status.Error(codes.NotFound, "Volume not found")
+		}
+		return 0, status.Errorf(codes.Internal, "Could not get volume with ID %q: %v", volumeID, err)
+	}
+
+	if val, ok := req.modifyTagsOptions.TagsToAdd[IopsPerGBKey]; ok {
+		iopsPerGb, err := strconv.ParseInt(val, 10, 32)
+		if err != nil {
+			return 0, fmt.Errorf("invalid iopspergb value %s", val)
+		}
+		if req.newSize != 0 {
+			req.modifyDiskOptions.IOPS = int32(iopsPerGb) * util.BytesToGiB(req.newSize)
+		} else {
+			req.modifyDiskOptions.IOPS = int32(iopsPerGb) * disk.CapacityGiB
+			req.newSize = util.GiBToBytes(disk.CapacityGiB) // Make sure to always pass in a size to capIOPS
+		}
+	}
+	return req.modifyDiskOptions.IOPS, nil
+}
```

Comment:
Why are we not just passing iopsPerGb to cloud? This causes us to double describe the volume